core/src/impl/Kokkos_Profiling.cpp

```diff
@@ -58,10 +64,136 @@
 #include <unordered_map>
 #include <unordered_set>
 #include <vector>
+#include <sstream>
+#include <iostream>
 namespace Kokkos {
 
 namespace Tools {
 
+InitArguments tool_arguments;
+
+const std::string InitArguments::unset_string_option = {
+    "kokkos_tools_impl_unset_option"};
+
+namespace Impl {
+void parse_command_line_arguments(int& narg, char* arg[],
+                                  InitArguments& arguments) {
+  int iarg = 0;
+  using Kokkos::Impl::check_arg;
+  using Kokkos::Impl::check_int_arg;
+  using Kokkos::Impl::check_str_arg;
```

Comment:
Where is `<impl/Kokkos_Command_Line_Parsing.hpp>` being included from?  It should be explicitly included here.

Reply:
It's included in the header, can include it in source too

Reply:
But it is only necessary in the source is it not?

Reply:
9f33083